src/main/java/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -80,10 +80,6 @@ class IonReaderContinuableApplicationBinary extends IonReaderContinuableCoreBina
     // The first lowest local symbol ID in the symbol table.
     private int firstLocalSymbolId = imports.getMaxId() + 1;
 
-    // A map of symbol ID to SymbolToken representation. Because most use cases only require symbol text, this
-    // is used only if necessary to avoid imposing the extra expense on all symbol lookups.
-    private List<SymbolToken> symbolTokensById = null;
```

Comment:
Another solution would be to change this to a `Map<Int, SymbolToken>` and then lazily populate it as symbols are requested. This means that (a) the size of the cache is determined by the number of unique `SymbolToken`s that are requested, and (b) each `SymbolToken` is allocated only once. However, it does incur the cost of looking something up in a `Map`.

Reply:
Yeah, I called this out in the PR description too. I'd prefer to remove the cache until it's clear that there is a use case that demands it.

Reply:
Wow, I don't know how I missed that.